### PR TITLE
feat: Add clearFilelist method

### PR DESCRIPTION
**Pull request by TatuLund**
fixes: https://github.com/vaadin/flow-components/issues/1572

### CODE REVIEW

File: vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java [hunk 1]
```diff
@@ -649,6 +650,13 @@ private String getStringObject(String propertyName, String object,
         return result;
     }
 
+    /**
+     * Clear the filelist of the component.
+     */
+    public void clearFilelist() {
```
Comment (sosa-vaadin) [inline vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java:656]:
```suggestion
    public void clearFileList() {
```

File: vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java [hunk 2]
```diff
@@ -649,6 +650,13 @@ private String getStringObject(String propertyName, String object,
         return result;
     }
 
+    /**
+     * Clear the filelist of the component.
```
Comment (sosa-vaadin) [inline vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java:654]:
```suggestion
     * Clear the fileList of the component.
```

**Review by sosa-vaadin — CHANGES_REQUESTED** [review]
Added suggestion to use camel case. It's a simple enough change otherwise.

**Review by sosa-vaadin — APPROVED** [review]

File: vaadin-upload-flow-parent/vaadin-upload-flow/src/test/java/com/vaadin/flow/component/upload/UploadClearFilesTest.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+public class UploadClearFilesTest {
+
+    String mockJson = "[{\"loaded\": 11037,\"held\": false,\"status\": \"\",\"xhr\": {},\"uploadTarget\": \"VAADIN/dynamic/resource/5/53877f5e-28d9-47a2-b061-80670615fb6e/upload\",\"formDataName\": \"file\",\"indeterminate\": false,\"uploading\": false,\"error\": false,\"abort\": false,\"complete\": true,\"progress\": 100}]";
+
+    private Upload upload;
+
+    @Before
+    public void initTest() throws IOException {
+        UI ui = new MockUI();
```
Comment (yuriy-fix) [inline vaadin-upload-flow-parent/vaadin-upload-flow/src/test/java/com/vaadin/flow/component/upload/UploadClearFilesTest.java:44]: I guess we can avoid adding `MockUI` and `MockVaadinSession` and instead create an integration test. (For example, modify `UploadView` and add test to `UploadIT`)

File: vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java (hunk 2 above)
Comment (tomivirkki) [reply vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java]: Why do we want to keep this as one word in the documentation? I don't think "fileList" is a known concept in `Upload`.

File: vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java [hunk 4]
```diff
@@ -649,6 +650,13 @@ private String getStringObject(String propertyName, String object,
         return result;
     }
 
+    /**
+     * Clear the fileList of the component.
+     */
+    public void clearFileList() {
+        getElement().setPropertyJson("files", Json.createArray());
```
Comment (tomivirkki) [inline vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java:657]: Use the `setFiles` method from the superclass (`GeneratedVaadinUpload`)

Comment (yuriy-fix) [reply vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java]: Updated.

**Review by yuriy-fix — APPROVED** [review]

File: vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java [hunk 5]
```diff
@@ -649,6 +650,13 @@ private String getStringObject(String propertyName, String object,
         return result;
     }
 
+    /**
+     * Clear the fileList of the component.
```
Comment (web-padawan) [inline vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java:654]:
```suggestion
     * Clear the list of files being processed, or already uploaded.
```

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**